src/api/environment.cc

```diff
@@ -91,6 +91,47 @@ static void HostCleanupFinalizationGroupCallback(
   env->RegisterFinalizationGroupForCleanup(group);
 }
 
+#define ATOMIC_WAIT_EVENTS(V)                                               \
+  V(kStartWait,           "started")                                        \
+  V(kWokenUp,             "was woken up by another thread")                 \
+  V(kTimedOut,            "timed out")                                      \
+  V(kTerminatedExecution, "was stopped by terminated execution")            \
+  V(kAPIStopped,          "was stopped through the embedder API")           \
+  V(kNotEqual,            "did not wait because the values mismatched")     \
+
+static void AtomicsWaitCallback(Isolate::AtomicsWaitEvent event,
+                                Local<v8::SharedArrayBuffer> array_buffer,
+                                size_t offset_in_bytes, int64_t value,
+                                double timeout_in_ms,
+                                Isolate::AtomicsWaitWakeHandle* stop_handle,
+                                void* data) {
+  Local<Context> context = array_buffer->CreationContext();
+  Environment* env = Environment::GetCurrent(context);
+  if (UNLIKELY(env == nullptr) ||
+      LIKELY(!env->options()->trace_atomics_wait)) {
+    return;
+  }
+
+  const char* message;
+  switch (event) {
+#define V(key, msg)                         \
+    case Isolate::AtomicsWaitEvent::key:    \
+      message = msg;                        \
+      break;
+    ATOMIC_WAIT_EVENTS(V)
+#undef V
+  }
+
+  fprintf(stderr,
+          "[Thread %" PRIu64 "] Atomics.wait(%p + %zx, %" PRId64 ", %.f) %s\n",
+          env->thread_id(),
+          array_buffer->GetBackingStore()->Data(),
+          offset_in_bytes,
+          value,
+          timeout_in_ms,
+          message);
```

Comment:
I'm a bit torn on the approach here. Alternatives would be:

* Use `Debug()` ala `NODE_DEBUG_NATIVE=ATOMICS_WAIT ./node`
* Use trace events

I think I'd prefer the former.

Reply:
> Use trace events

I’d prefer not to do that, the output here should be easily consumable by developers. 

The downside to the `Debug()` approach is that `NODE_DEBUG_NATIVE` is intended to be used by Node.js core developers, not by application developers, and we’re (rightfully imo) not advertising that environment variable very loudly.

Reply:
I've been leaning in favor of a `Debug` that *is* intended for end users to reduce the number of command line flags we need to add (and maintain). Especially given that `DEBUG` is already a pattern that is accepted and used in the wild modeled after our `NODE_DEBUG` (https://www.npmjs.com/package/debug). Consider this non-blocking for now, however.